man/fabtests.7.md

```diff
@@ -0,0 +1,113 @@
+
+## OVERVIEW
+  
+Libfabric defines sets of interface that fabric providers can support. The purpose of Fabtests examples is to demonstrate some of the major features. The goal is to familiarize users with different functionalities libfabric offers and how to use them.
+
+The tests are divided into four categories:
+
+### Simple
+
+These tests are a mix of very basic tests and major features of libfabric. All of the tests except info.c are designed to run as client-server processes. A server is started first and then a client process connects to the server and performs various operations.
+
+	/simple/fi_cq_data: A client-server example that tranfers CQ data
+	/simple/fi_dgram: A basic DGRAM client-server example
+	/simple/fi_dgram_waitset: A basic DGRAM client-server example that uses waitset
+	/simple/fi_info: An example (non client-server) that prints fabric interface information obtained by fi_getinfo call
+	/simple/fi_msg: A basic MSG client-server example
+	/simple/fi_msg_pingpong: A ping-pong client-server example using MSG endpoints
+	/simple/fi_msg_rma: A ping pong client-server example using RMA operations between MSG endpoints
+	/simple/fi_poll: A basic RDM client-server example that uses poll
+	/simple/fi_rdm: A basic RDM client-server example
+	/simple/fi_rdm_atomic: An RDM ping pong client-server using atomic operations
+	/simple/fi_rdm_cntr_pingpong: An RDM ping pong client-server using counters
+	/simple/fi_rdm_inject_pingpong: An RDM ping pong client-server example using inject
+	/simple/fi_rdm_multi_recv: An RDM ping pong client-server example using multi recv buffer
+	/simple/fi_rdm_pingpong: A ping pong client-server example using RDM endpoints
+	/simple/fi_rdm_rma: A ping pong client-server example using RMA operations
+	/simple/fi_rdm_rma_simple: A simple RDM client-sever RMA example
+	/simple/fi_rdm_shared_ctx: An RDM client-server example that uses shared context
+	/simple/fi_rdm_tagged_pingpong: A ping pong client-server example using tagged messages
+	/simple/fi_rdm_tagged_search: An RDM client-server example that uses tagged search
+	/simple/fi_scalable_ep: An RDM client-server example with scalable endpoints
+	/simple/fi_ud_pingpong: A ping-pong client-server example using DGRAM endpoints
+
+### Unit
+	 /unit/fi_eq_test
+	 /unit/fi_dom_test
+	 /unit/fi_av_test
+	 /unit/fi_size_left_test
+
+### Ported
+	 /ported/librdmacm/fi_cmatose
+	 /ported/libibverbs/fi_rc_pingpong
+
+### Complex
+	 Under development
+
+## HOW TO RUN TESTS
+(1) You need to build libfabric before running fabtests.
+
+	export LD_LIBRARY_PATH=/path/to/libfabric/install:$LD_LIBRARY_PATH
+
+(2) Follow README to build fabtests.
+
+(3) All the test executables are prefixed with "fi_". For example, in order to run simple/rdm_pingpong.c, the test executable will be simple/fi_rdm_pingpong.
+
+All the tests except fi_info in the simple folder have the following usage model and minimum options:
```

Comment:
Replace 'all' with 'most'.  For example, the unit tests do not follow this model either.